### ISSUE TITLE
Trouble with Revit 2019

### ISSUE THREAD

**Issue by sebacos** [va3c/RvtVa3c#15]
Hey guys,

I followed the instructions on the front page.
I have been trying to add the add-in to Revit 2019 and while yesterday I manage to add it to Revit 2018 it is nog working now and Visual Studio gives me following errors:
I am not a programmer, so my knowlage to repair something is kind of limited.

Hope you can help me out. 

Thank you kindly.

`1>------ Build started: Project: DynVa3c, Configuration: Debug Any CPU ------
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3274: The primary reference "RvtVa3c" could not be resolved because it was built against the ".NETFramework,Version=v4.7.2" framework. This is a higher version than the currently targeted framework ".NETFramework,Version=v4.5".
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "DSCoreNodes". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "DynamoCore". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "ProtoCore". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "ProtoGeometry". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "ProtoInterface". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "RevitAPI". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "RevitAPIUI". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "RevitNodes". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2106,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "RevitServices". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(8,7,8,15): error CS0246: The type or namespace name 'Autodesk' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(9,7,9,15): error CS0246: The type or namespace name 'Autodesk' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(10,7,10,15): error CS0246: The type or namespace name 'Autodesk' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(11,7,11,13): error CS0246: The type or namespace name 'DSCore' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(12,7,12,13): error CS0246: The type or namespace name 'Dynamo' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(13,7,13,13): error CS0246: The type or namespace name 'Dynamo' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(14,7,14,13): error CS0246: The type or namespace name 'Dynamo' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(15,7,15,16): error CS0246: The type or namespace name 'ProtoCore' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(16,7,16,20): error CS0246: The type or namespace name 'RevitServices' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(17,7,17,20): error CS0246: The type or namespace name 'RevitServices' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(18,7,18,16): error CS0246: The type or namespace name 'ProtoCore' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(19,7,19,16): error CS0246: The type or namespace name 'ProtoCore' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(20,7,20,15): error CS0246: The type or namespace name 'Autodesk' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(21,7,21,12): error CS0246: The type or namespace name 'Revit' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs(22,7,22,14): error CS0246: The type or namespace name 'RvtVa3c' could not be found (are you missing a using directive or an assembly reference?)
========== Build: 0 succeeded, 1 failed, 1 up-to-date, 0 skipped ==========`

Severity	Code	Description	Project	File	Line	Suppression State	Suppression State
Error	CS0246	The type or namespace name 'Autodesk' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	8		Active
Error	CS0246	The type or namespace name 'Autodesk' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	9		Active
Error	CS0246	The type or namespace name 'Autodesk' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	10		Active
Error	CS0246	The type or namespace name 'DSCore' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	11		Active
Error	CS0246	The type or namespace name 'Dynamo' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	12		Active
Error	CS0246	The type or namespace name 'Dynamo' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	13		Active
Error	CS0246	The type or namespace name 'Dynamo' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	14		Active
Error	CS0246	The type or namespace name 'ProtoCore' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	15		Active
Error	CS0246	The type or namespace name 'RevitServices' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	16		Active
Error	CS0246	The type or namespace name 'RevitServices' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	17		Active
Error	CS0246	The type or namespace name 'ProtoCore' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	18		Active
Error	CS0246	The type or namespace name 'ProtoCore' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	19		Active
Error	CS0246	The type or namespace name 'Autodesk' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	20		Active
Error	CS0246	The type or namespace name 'Revit' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	21		Active
Error	CS0246	The type or namespace name 'RvtVa3c' could not be found (are you missing a using directive or an assembly reference?)	DynVa3c	C:\Users\AS\Downloads\RvtVa3c-gh-pages (1)\RvtVa3c-gh-pages\DynVa3c\Va3c4Dynamo.cs	22		Active
Warning		Could not resolve this reference. Could not locate the assembly "ProtoGeometry". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.	DynVa3c				
Warning		Could not resolve this reference. Could not locate the assembly "DSCoreNodes". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.	DynVa3c				
Warning		The referenced component 'DSCoreNodes' could not be found.	DynVa3c				
Warning		The referenced component 'RvtVa3c' could not be found.	DynVa3c				
Warning		The referenced component 'DynamoCore' could not be found.	DynVa3c				
Warning		Could not resolve this reference. Could not locate the assembly "ProtoInterface". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.	DynVa3c				
Warning		The referenced component 'ProtoInterface' could not be found.	DynVa3c				
Warning		The referenced component 'RevitServices' could not be found.	DynVa3c				
Warning		Could not resolve this reference. Could not locate the assembly "RevitNodes". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.	DynVa3c				
Warning		The primary reference "RvtVa3c" could not be resolved because it was built against the ".NETFramework,Version=v4.7.2" framework. This is a higher version than the currently targeted framework ".NETFramework,Version=v4.5".	DynVa3c				
Warning		The referenced component 'RevitAPIUI' could not be found.	DynVa3c				
Warning		The referenced component 'ProtoCore' could not be found.	DynVa3c				
Warning		Could not resolve this reference. Could not locate the assembly "ProtoCore". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.	DynVa3c				
Warning		The referenced component 'RevitNodes' could not be found.	DynVa3c				
Warning		Could not resolve this reference. Could not locate the assembly "RevitAPI". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.	DynVa3c				
Warning		Could not resolve this reference. Could not locate the assembly "RevitAPIUI". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.	DynVa3c				
Warning		Could not resolve this reference. Could not locate the assembly "RevitServices". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.	DynVa3c				
Warning		Could not resolve this reference. Could not locate the assembly "DynamoCore". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.	DynVa3c				
Warning		The referenced component 'RevitAPI' could not be found.	DynVa3c				
Warning		The referenced component 'ProtoGeometry' could not be found.	DynVa3c

**Comment by jeremytammik** [comment]
The version of RvtVa3c that you are attempting to build is set up for Revit 2018, which uses the .NET framework version 4.5. The Revit 2019 API uses the .NET framework version 4.7. You need to modify the target .NET framework version in your build settings.

**Comment by jeremytammik** [comment]
I'm afraid I do not have Revit 2019 installed. However, I migrated RvtVa3c to Revit 2020 for you. It also uses the .NET 4.7 framework, so the changes I made should be identical for Revit 2o19 as well. The flat migration from Revit 2018 to Revit 2020 was completely trivial. All I did was update the .NET target framework version number from 4.5 to 4.7 and update the Revit API references to RevitAPI.dll and RevitUIAPI.dll to the new Revit 2020 directory path.
